src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -113,6 +113,16 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode>, pr
   std::shared_ptr<AbstractLQPNode> child(LQPChildSide side) const;
 
   void set_child(LQPChildSide side, const std::shared_ptr<AbstractLQPNode>& child);
+
+  /*
+   * @returns the number of children
+   */
+  uint8_t child_count() const;
```

Comment:
minor: for simplicity I'd rather have anything involving "counts" return size_t or equivalent. Also, if you're unlucky and you add/multiply/... function results that all "because it's sufficient" return uint8_t you might get integer overflows. Let's not take chances here, imo.